src/spec/expectations.cr

```diff
@@ -417,6 +417,26 @@ module Spec
   end
 
   module ObjectExtensions
+    # Validates an expectation and fails the example if it does not match.
+    #
+    # This overload will be called when doing:
+    #
+    # ```
+    # result = subject.should be_a(Type)
+    # # result will be a Type
```

Comment:
Describing this in code is IMO clearer:
```suggestion
    # typeof(result) # => Type
```

Even better to use a working example (i.e. copied from the spec):
```cr
x = 1 || 'a'
typeof(x) # => Int32 | Char
x = x.should be_a(Int32)
typeof(x) # => Int32
```
This is certainly easier to understand for the `should_not` matcher.

Reply:
Great idea! I did this for all the overloads. Let me know what you think.